src/components/DayPicker.jsx

```diff
@@ -71,8 +75,10 @@ const defaultProps = {
 export default class DayPicker extends React.Component {
   constructor(props) {
     super(props);
+
+    this.hasSetInitialVisibleMonth = !!props.focused;
     this.state = {
-      currentMonth: moment(),
+      currentMonth: props.focused ? props.initialVisibleMonth() : moment(),
```

Comment:
something like `const triggerInitialVisibleMonth = (props) => props.focused ? props.initialVisibleMonth() : moment();` for clarity?